packages/@react-aria/list/docs/useList.mdx

```diff
@@ -0,0 +1,439 @@
+import {Checkbox} from 'your-component-library';
+
+function ListCheckbox({ item, state }) {
+  let ref = useRef();
+  let { checkboxProps } = useListSelectionCheckbox({ key: item.key }, state);
+
+  return (
+    <span ref={ref}>
+      <Checkbox {...checkboxProps} />
+    </span>
+  );
+}
+```
+
+### Checkbox
+
+The `Checkbox` component is used in the above example for row selection. It is built using the [useCheckbox](useCheckbox.html) hook, and can be shared with many other components.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+import {useCheckbox} from '@react-aria/checkbox';
+import {useToggleState} from '@react-stately/toggle';
+
+function Checkbox(props) {
+  let inputRef = useRef(null);
+  let { inputProps } = useCheckbox(
+    props,
+    useToggleState(props),
+    inputRef
+  );
+  return <input {...inputProps} ref={inputRef} />;
+}
+```
+
+</details>
+
+### Button
+
+The `Button` component is used in the above example to show how rows can contain interactive elements. It is built using the [useButton](useButton.html) hook, and can be shared with many other components.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+import {useButton} from '@react-aria/button';
+
+function Button(props) {
+  let ref = React.useRef();
+  let {buttonProps} = useButton(props, ref);
+  return <button {...buttonProps} ref={ref}>{props.children}</button>;
+}
+```
+
+</details>
+
+The following example shows how to enable multiple selection support using the List component we built above.
+It's as simple as setting the `selectionMode` prop to `"multiple"`.
+
+```tsx example
+<List aria-label="List with selection" selectionMode="multiple">
+  <Item key="1">Charizard</Item>
+  <Item key="2">Blastoise</Item>
+  <Item key="3">Venusaur</Item>
+  <Item key="4">Pikachu</Item>
+</List>
+```
+
+And that's it! We now have a fully interactive List component that can support keyboard navigation, single or multiple selection.
+In addition, it is fully accessible for screen readers and other assistive technology. See below for more
+examples of how to use the List component that we've built.
+
+## Usage
+
+### Dynamic collections
+
+So far, our examples have shown static collections, where the data is hard coded.
+Dynamic collections, as shown below, can be used when the List data comes from an external data source such as an API, or updates over time.
+In the example below, the rows are provided to the List via a render function.
+
+```tsx example export=true
+function ExampleList(props) {
+  let rows = [
+    {id: 1, name: 'Games'},
+    {id: 2, name: 'Program Files'},
+    {id: 3, name: 'bootmgr'},
+    {id: 4, name: 'log.txt'}
+  ];
+
+  return (
+    <List aria-label="Example dynamic collection List" selectionMode="single" items={rows} {...props}>
```

Comment:
```suggestion
    <List aria-label="Example dynamic collection List" selectionMode="multiple" items={rows} {...props}>
```

Single selection with checkboxes is weird, maybe keep single selection to highlight, or switch to multi for checkboxes.